langmeta-sqlite/src/main/scala/org/langmeta/sqlite/Main.scala

```diff
@@ -0,0 +1,214 @@
+                        symbolStmt.setInt(1, symbolRef)
+                        symbolStmt.setString(2, symbol.symbol)
+                        symbolStmt.setLong(3, symbol.denotation.get.flags)
+                        symbolStmt.setString(4, symbol.denotation.get.name)
+                        val signatureDocumentRef = {
+                          documentStmt.setInt(1, documentId.next)
+                          documentStmt.setString(2, null)
+                          documentStmt.setString(3, symbol.denotation.get.signature)
+                          documentStmt.setString(4, null)
+                          documentStmt.executeUpdate()
+                          documentId.value
+                        }
+                        symbolStmt.setInt(5, signatureDocumentRef)
+                        symbol.denotation.get.names.foreach { name =>
+                          nameStmt.setInt(1, nameId.next)
+                          nameStmt.setInt(2, signatureDocumentRef)
+                          nameStmt.setInt(3, name.position.get.start)
+                          nameStmt.setInt(4, name.position.get.end)
+                          nameStmt.setInt(5, symbolId(name.symbol))
+                          nameStmt.setBoolean(6, name.isDefinition)
+                          nameStmt.executeUpdate()
+                        }
+                        symbolStmt.executeUpdate()
+                        symbolTodo.remove(symbolRef)
+                        symbolDone.add(symbolRef)
+                      }
+                    }
+
+                    document.synthetics.foreach { synthetic =>
+                      syntheticStmt.setInt(1, syntheticId.next)
+                      syntheticStmt.setInt(2, synthetic.pos.get.start)
+                      syntheticStmt.setInt(3, synthetic.pos.get.end)
+                      val syntheticDocumentRef = {
+                        documentStmt.setInt(1, documentId.next)
+                        documentStmt.setString(2, null)
+                        documentStmt.setString(3, synthetic.text)
+                        documentStmt.setString(4, null)
+                        documentStmt.executeUpdate()
+                        documentId.value
+                      }
+                      syntheticStmt.setInt(4, syntheticDocumentRef)
+                      synthetic.names.foreach { name =>
+                        nameStmt.setInt(1, nameId.next)
+                        nameStmt.setInt(2, syntheticDocumentRef)
+                        nameStmt.setInt(3, name.position.get.start)
+                        nameStmt.setInt(4, name.position.get.end)
+                        nameStmt.setInt(5, symbolId(name.symbol))
+                        nameStmt.setBoolean(6, name.isDefinition)
+                        nameStmt.executeUpdate()
+                      }
+                      syntheticStmt.executeUpdate()
+                    }
+
+                    if ((genuineDocuments % 1000) == 0) {
```

Comment:
The logic in this block also needs to be called after the main loop exits, since we can't expect that the number of documents is going to always divide by 1000.